app/models/location.rb

```diff
@@ -199,6 +201,57 @@ class Location < AbstractModel # rubocop:disable Metrics/ClassLength
     )
   }
 
+  scope :contains, # Use named parameters (lat:, lng:, or n:, s:, e:, w:)
+        lambda { |**args|
+          if args[:lat] && args[:lng]
+            args => {lat:, lng:}
+            n = s = lat
+            e = w = lng
+          elsif args[:n]
+            args => {n:, s:, e:, w:}
+          end
+
+          # Correct for possible floating point rounding
+          shrunk_n = n - ROUND_ERROR
+          shrunk_s = s + ROUND_ERROR
+          shrunk_e = e - ROUND_ERROR
+          shrunk_w = w + ROUND_ERROR
+
+          # w/e    | Location     | Location contains w/e
+          # ______ | ____________ | ______________________
+          # w <= e | west <= east | west <= w && e <= east
+          # w <= e | west > east  | west <= w || e <= east
+          # w > e  | west <= east | none
+          # w > e  | west > east  | west <= w && e <= east
+
+          if w <= e
+            # w / e don't straddle 180
+            where(Location[:south].lteq(shrunk_s).
+                    and(Location[:north].gteq(shrunk_n)).
+                  #   Location doesn't straddle 180
+                  and(Location[:west].lteq(Location[:east]).
+                      and(Location[:west] <= shrunk_w).
+                      and(Location[:east] >= shrunk_e).
+                    #  Location straddles 180
+                    or(Location[:west].gt(Location[:east]).
+                      #    need not check e, since w <= e
+                      and((Location[:west] <= shrunk_w).
+                        #  need not check w for same reason
+                        or(Location[:east] >= shrunk_e)))))
+          else
+            # w / e straddle 180
+            where(Location[:south].lteq(shrunk_s).
+                  and(Location[:north].gteq(shrunk_n)).
+            # Location straddles 180
+            #   Location 100% wrap; necessarily straddles w/e
+            and(Location[:west] == Location[:east] - 360).
+            #   Location < 100% wrap-around
+            or((Location[:west].gt(Location[:east])). # rubocop:disable Style/RedundantParentheses
+              and(Location[:west] <= shrunk_w).
+              and(Location[:east] >= shrunk_e)))
+          end
+        }
```

Comment:
@nimmolo: This is complex because the conditions that work for Location.contains(point)
```ruby
              Location[:west].lteq(lng).and(Location[:east].gteq(lng)).or(
                Location[:west].gteq(lng).and(Location[:east].lteq(lng))
```
do not work in 2 dimensions. For instance, Alaska would not contains most places inside Alaska.

@nimmolo & @mo-nathan: I would appreciate anything that would simplify this mess, or even just make it easier to understand or read.

Reply:
Looks pretty good, not sure how it could be improvable. Ideas:
- Maybe let's split the one scope into two: a `contains_point` and a `contains_box`. That will help legibility at least and shorten the scope.